### PR TITLE
3.0.21 Improvements and bug fix.

### PR DESCRIPTION
**Pull request by jonhy81**
* MULTISITE-20750: Set default value for drush.color as 0.
  * MULTISITE-21001: Improve target platform-resource-link to check if files exists before creation.
  * MULTISITE-21025: Improvements to default behat configuration.
  * MULTIISTE-21027: Fix install-clean for multisite_drupal_communities profile.
  * MULTISITE-21046: Set CSP during settings generation for install-clean and install-clone.
  * MULTISITE-21052: Store database dump in compress format (.gz).
  * MULTISITE-21073: Make toolkit compatible with PHP 7.1.
  * MULTISITE-21121: Bugfix, SOLR environment now set for communities profile.
  * MULTISITE-21227: Force delete of dist folder during distribution rebuild.
  * MULTISITE-21234: Remove dependency on RELEASE endpoint, platform fetched from GitHub.
  * MULTISITE-21260: Automatic recover uid:1 if table users is empty.
  * OTHERS: Improve documentation for Supported profiles.

### CODE REVIEW

File: build.default.props [hunk 1]
```diff
@@ -178,7 +178,6 @@ behat.wd_host.url = http://selenium:4444/wd/hub
 
 # Binaries.
 # ---------
-project.bin.composer = composer.phar
```
Comment (ghost) [inline build.default.props]: @jonhy81 could you tell me why this was done? My local development system relies on being able to change the composer binary location...

File: includes/phing/src/Tasks/PlatformVersionsTask.php [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+        ksort($versions);
+
+        // Check if end-user is providing the exact version, if not just get the latest
+        // for the major provided.
+        if (in_array($this->_packageVersion, $versions)) {
+            $this->setVersionProp($this->_packageVersion);
+        }
+        else {
+            foreach($versions as $version) {
+                $temporaryGroups[substr_compare($version, $this->_majorVersion, 0, 3)] = $version;
```
Comment (ghost) [inline includes/phing/src/Tasks/PlatformVersionsTask.php:99]: I think this is responsible for the breaking of platform version 2.3 in the toolkit pipeline:
https://webgate.ec.europa.eu/CITnet/jira/browse/MULTISITE-21403

I'm not sure how to solve this issue. It needs to be solved if we can't guarantee that the platform won't release other packages like this.

Comment (ghost) [reply includes/phing/src/Tasks/PlatformVersionsTask.php:99]: Fixed https://github.com/ec-europa/toolkit/pull/146/commits/fb7eb1b6cb799d7351eb14fa5bbd4744e61ec8a9

File: build.default.props (hunk 1 above)
Comment (jonhy81) [reply build.default.props]: `project.bin.composer` back to toolkit.

**Review by ghost — APPROVED** [review]

**Pull request merged**